File: SDKs/asana.com/1.0/python/src/sdk/models/operations/removeuserforteam.py
from dataclasses import dataclass, field
from typing import Any,List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared


@dataclass
class RemoveUserForTeamPathParams:
    team_gid: str = field(default=None, metadata={'path_param': { 'field_name': 'team_gid', 'style': 'simple', 'explode': False }})
    

@dataclass
class RemoveUserForTeamQueryParams:
    opt_fields: Optional[List[str]] = field(default=None, metadata={'query_param': { 'field_name': 'opt_fields', 'style': 'form', 'explode': False }})
    opt_pretty: Optional[bool] = field(default=None, metadata={'query_param': { 'field_name': 'opt_pretty', 'style': 'form', 'explode': True }})
    

@dataclass_json
@dataclass
class RemoveUserForTeamRequestBody:
    data: Optional[shared.TeamRemoveUserRequest] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'data' }})
    

@dataclass
class RemoveUserForTeamRequest:
    path_params: RemoveUserForTeamPathParams = field(default=None)
    query_params: RemoveUserForTeamQueryParams = field(default=None)
    request: RemoveUserForTeamRequestBody = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass_json
@dataclass
class RemoveUserForTeam204ApplicationJSON:
    data: Optional[dict[str, Any]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'data' }})
    

@dataclass
class RemoveUserForTeamResponse:
    content_type: str = field(default=None)
    error_response: Optional[shared.ErrorResponse] = field(default=None)
    status_code: int = field(default=None)
    remove_user_for_team_204_application_json_object: Optional[RemoveUserForTeam204ApplicationJSON] = field(default=None)
    

File: SDKs/asana.com/1.0/typescript/src/sdk/models/operations/removeuserforteam.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class RemoveUserForTeamPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=team_gid" })
  teamGid: string;
}


export class RemoveUserForTeamQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=opt_fields" })
  optFields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=opt_pretty" })
  optPretty?: boolean;
}


export class RemoveUserForTeamRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: shared.TeamRemoveUserRequest;
}


export class RemoveUserForTeam204ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: Map<string, any>;
}


export class RemoveUserForTeamRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: RemoveUserForTeamPathParams;

  @SpeakeasyMetadata()
  queryParams: RemoveUserForTeamQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: RemoveUserForTeamRequestBody;
}


export class RemoveUserForTeamResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorResponse?: shared.ErrorResponse;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  removeUserForTeam204ApplicationJsonObject?: RemoveUserForTeam204ApplicationJson;
}
